src/coreclr/jit/emit.cpp

```diff
@@ -1007,8 +1007,14 @@ insGroup* emitter::emitSavIG(bool emitAdd)
         VarSetOps::Assign(emitComp, emitPrevGCrefVars, emitThisGCrefVars);
         emitPrevGCrefRegs = emitThisGCrefRegs;
         emitPrevByrefRegs = emitThisByrefRegs;
+    }
 
+    if (emitAddedLabel)
```

Comment:
I would leave this whole `if` within the above `if (!emitAdd)` clause. Though it probably doesn't matter.

Reply:
ah, makes sense. We might not want to do it for extended blocks.